front/pages/w/[wId]/builder/data-sources/[name]/index.tsx

```diff
@@ -262,6 +265,47 @@ function DatasourceDocumentsTabView({
   const [displayNameByDocId, setDisplayNameByDocId] = useState<
     Record<string, string>
   >({});
+  const sendNotification = useContext(SendNotificationsContext);
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const [bulkFilesUploading, setBulkFilesUploading] = useState<null | {
+    total: number;
+    completed: number;
+  }>(null);
+
+  const handleUpsert = async (text: string, documentId: string) => {
+    const body: PostDataSourceDocumentRequestBody = {
+      timestamp: null,
+      parents: null,
+      section: {
+        prefix: null,
+        content: text,
+        sections: [],
+      },
+      text: null,
+      source_url: undefined,
+      tags: [],
+      light_document_output: true,
+      upsert_context: null,
+      async: false,
+    };
+
+    const res = await fetch(
+      `/api/w/${owner.sId}/data_sources/${
+        dataSource.name
+      }/documents/${encodeURIComponent(documentId)}`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(body),
+      }
+    );
+
+    if (!res.ok) {
+      throw new Error("Error uploading document.");
```

Comment:
This will swallow the error message and will make a notification that is not indicative. Esp when going over the data source limit.

+ since we try to catch the error outside this function we want to use a Result here as per our error handling convention